dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -256,83 +264,102 @@ public Workflow getWorkflow(String repositoryId, Optional<Workflow> existingWork
-                    sourceFiles = tag.getSourceFiles();
-                    if (type == DescriptorLanguage.CWL) {
-                        filePath = tag.getCwlPath();
-                        version = tag;
-                    } else if (type == DescriptorLanguage.WDL) {
-                        filePath = tag.getWdlPath();
-                        version = tag;
-                    } else {
-                        throw new UnsupportedOperationException("tool is not a CWL or WDL file");
-                    }
+            Tool tool = (Tool)entry;
+            tool.getWorkflowVersions().forEach(tag -> {
+                String filePath;
+                if (type == DescriptorLanguage.CWL) {
+                    filePath = tag.getCwlPath();
+                } else if (type == DescriptorLanguage.WDL) {
+                    filePath = tag.getWdlPath();
+                } else {
+                    throw new UnsupportedOperationException("tool is not a CWL or WDL file");
                 }
-            }
+                updateVersionMetadata(filePath, tag, type, repositoryId);
+            });
         }
-
-        // If entry is a workflow
-
         if (entry instanceof Workflow) {
-            // Find filepath to parse
-            for (WorkflowVersion workflowVersion : ((Workflow)entry).getWorkflowVersions()) {
-                if (workflowVersion.getReference().equals(branch)) {
-                    filePath = workflowVersion.getWorkflowPath();
-                    sourceFiles = workflowVersion.getSourceFiles();
-                    version = workflowVersion;
-                }
+            Workflow workflow = (Workflow)entry;
+            workflow.getWorkflowVersions().forEach(workflowVersion -> {
+                String filePath = workflowVersion.getWorkflowPath();
+                updateVersionMetadata(filePath, workflowVersion, type, repositoryId);
+            });
+        }
+        return entry;
+    }
+
+    /**
+     * Sets the default version if there isn't already one present.
+     * This is required because entry-level metadata depends on the default version
+     *
+     * @param entry
+     * @param repositoryId
+     */
+    public void setDefaultBranch(Entry entry, String repositoryId) {
+        if (entry.getDefaultVersion() == null) {
+            String branch = getMainBranch(entry, repositoryId);
+            if (branch == null) {
+                String message = String.format("%s : Error getting the main branch.", repositoryId);
+                LOG.info(message);
+            } else {
+                Set<Version> workflowVersions = entry.getWorkflowVersions();
+                Optional<Version> firstWorkflowVersion = workflowVersions.stream()
+                        .filter(workflowVersion -> {
+                            String reference = workflowVersion.getReference();
+                            return reference != null && reference.equals(branch);
```

Comment:
You could do `branch.equals(reference)`, then you don't have to do a null check.